### PR TITLE
Fix: Redis Readiness Check with password

### PR DESCRIPTION
**Pull request by Chickensoupwithrice**
Should resolve https://github.com/sourcegraph/deploy-sourcegraph-helm/issues/457

### Checklist

- [x] Follow the [manual testing process](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/TEST.md)
- [x] Update [changelog](https://github.com/sourcegraph/deploy-sourcegraph-helm/blob/main/charts/sourcegraph/CHANGELOG.md)
- [ ] Update [Kubernetes update doc](https://docs.sourcegraph.com/admin/updates/kubernetes)

### Test plan

Tested locally

### CODE REVIEW

File: charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml [hunk 1]
```diff
@@ -67,6 +67,9 @@ spec:
             - -c
             - |
               #!/bin/bash
+              if [ -f /etc/redis/redis.conf ]; then
+                REDISCLI_AUTH=$(grep --no-filename --no-heading --no-line-number "requirepass" /etc/redis/redis.conf | \tr -s ' '| cut -d ' ' -f 3)
```
Comment (Chickensoupwithrice) [inline charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml:71]: I'm quickly becoming the dangerous bash one-liner guy, huh? 

I don't like this branding...

**Comment by Chickensoupwithrice** [comment]
Tested with no password: 

image

**Comment by Chickensoupwithrice** [comment]
Tested with password: 

image

Full testing patch: 

```
diff --git a/charts/sourcegraph/templates/redis/redis-cache.ConfigMap.yaml b/charts/sourcegraph/templates/redis/redis-cache.ConfigMap.yaml
new file mode 100644
index 0000000..56a706a
--- /dev/null
+++ b/charts/sourcegraph/templates/redis/redis-cache.ConfigMap.yaml
@@ -0,0 +1,27 @@
+apiVersion: v1
+data:
+  redis.conf: |
+    #############################
+    ## Default redis.conf file           ##
+    #############################
+    # allow access from all instances
+    protected-mode no
+    # limit memory usage, discard unused keys when hitting limit
+    maxmemory 6gb
+    maxmemory-policy allkeys-lru
+    # snapshots on disk every minute
+    dir /redis-data/
+    appendonly no
+    save 60 1
+
+    ###################
+    ## Customization ##
+    ###################
+    requirepass demopasswordchangeme123
+
+kind: ConfigMap
+metadata:
+  labels:
+    app.kubernetes.io/component: redis-cache
+    deploy: sourcegraph
+  name: redis-cache-conf
diff --git a/charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml b/charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml
index 92a9d5b..e89ac43 100644
--- a/charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml
+++ b/charts/sourcegraph/templates/redis/redis-cache.Deployment.yaml
@@ -94,6 +94,8 @@ spec:
         volumeMounts:
         - mountPath: /redis-data
           name: redis-data
+        - mountPath: /etc/redis/
+          name: redis-cache-conf
         {{- if .Values.redisCache.extraVolumeMounts }}
         {{- toYaml .Values.redisCache.extraVolumeMounts | nindent 8 }}
         {{- end }}
@@ -132,6 +134,9 @@ spec:
       - name: redis-data
         persistentVolumeClaim:
           claimName: redis-cache
+      - configMap:
+          name: redis-cache-conf
+        name: redis-cache-conf
       {{- if .Values.redisCache.extraVolumes }}
       {{- toYaml .Values.redisCache.extraVolumes | nindent 6 }}
       {{- end }}
diff --git a/charts/sourcegraph/templates/redis/redis-store.ConfigMap.yaml b/charts/sourcegraph/templates/redis/redis-store.ConfigMap.yaml
new file mode 100644
index 0000000..fff4eb3
--- /dev/null
+++ b/charts/sourcegraph/templates/redis/redis-store.ConfigMap.yaml
@@ -0,0 +1,27 @@
+apiVersion: v1
+data:
+  redis.conf: |
+    #############################
+    ## Default redis.conf file           ##
+    #############################
+    # allow access from all instances
+    protected-mode no
+    # limit memory usage, discard unused keys when hitting limit
+    maxmemory 6gb
+    maxmemory-policy allkeys-lru
+    # snapshots on disk every minute
+    dir /redis-data/
+    appendonly no
+    save 60 1
+
+    ###################
+    ## Customization ##
+    ###################
+    requirepass demopasswordchangeme123
+
+kind: ConfigMap
+metadata:
+  labels:
+    app.kubernetes.io/component: redis-store
+    deploy: sourcegraph
+  name: redis-store-conf
diff --git a/charts/sourcegraph/templates/redis/redis-store.Deployment.yaml b/charts/sourcegraph/templates/redis/redis-store.Deployment.yaml
index f9f8188..6546213 100644
--- a/charts/sourcegraph/templates/redis/redis-store.Deployment.yaml
+++ b/charts/sourcegraph/templates/redis/redis-store.Deployment.yaml
@@ -94,6 +94,8 @@ spec:
         volumeMounts:
         - mountPath: /redis-data
           name: redis-data
+        - mountPath: /etc/redis/
+          name: redis-store-conf
         {{- if .Values.redisStore.extraVolumeMounts }}
         {{- toYaml .Values.redisStore.extraVolumeMounts | nindent 8 }}
         {{- end }}
@@ -132,6 +134,9 @@ spec:
       - name: redis-data
         persistentVolumeClaim:
           claimName: redis-store
+      - configMap:
+          name: redis-store-conf
+        name: redis-store-conf
       {{- if .Values.redisStore.extraVolumes }}
       {{- toYaml .Values.redisStore.extraVolumes | nindent 6 }}
       {{- end }}
```

**Review by daxmc99 — APPROVED** [review]
LGTM

**Review by jdpleiness — APPROVED** [review]
I think this is a pretty safe one-liner 😆 
LGTM

**Pull request merged**